buildroot-external/patches/occu/0154-WebUI-DynamicResizing.patch

```diff
@@ -0,0 +1,333 @@
+-
+-    if (width  < this.MIN_WIDTH)  { width  = this.MIN_WIDTH;  bodyOverflow = "auto"; }
+-    if (height < this.MIN_HEIGHT) { height = this.MIN_HEIGHT; bodyOverflow = "auto"; }
+-    var contentHeight = height - this.STATIC_HEIGHT;
+-
+-    if ($("body"))    { Element.setStyle("body", {"overflow": bodyOverflow, "width": width  + "px", "height": height + "px"}); }
+-    if ($("header"))  { Element.setStyle("header" , {"height": this.HEADER_HEIGHT  + "px", "width": width + "px"}); }
+-    if ($("menubar")) { Element.setStyle("menubar", {"height": this.MENUBAR_HEIGHT + "px", "width": width + "px"}); }
+-    if ($("content")) { Element.setStyle("content", {"height": contentHeight       + "px", "width": width + "px"}); }
+-    if ($("footer"))  { Element.setStyle("footer" , {"height": this.FOOTER_HEIGHT  + "px", "width": width + "px"}); }
+-
+     if (this.currentPage) { this.currentPage.resize(); }
+-
+-    if(typeof dcTimeout == "undefined") {
+-      dcTimeout = window.setTimeout(function () {
+-        showDutyCycle();
+-        delete dcTimeout;
+-      }, 10);
+-    }
+   },
+ 
+   /*########################*/
+@@ -20190,19 +20169,6 @@
+   initialize: function ()
+   {
+     this.deviceList = [];
+-
+-    switch (getUPL())
+-    {
+-      case UPL_ADMIN:
+-      case UPL_USER:
+-        this.resize = this.resizeUser;
+-        break;
+-      case UPL_GUEST:
+-        this.rezise = this.resizeGuest;
+-        break;
+-      default:
+-        break;
+-    }
+   },
+ 
+   /**
+@@ -20229,35 +20195,6 @@
+         break;
+     }
+   },
+-  
+-  /**
+-   * Berechnet die Gr��e der normalen Startseite (Admin und Benutzer)
+-   **/
+-  resizeUser: function()
+-  {
+-    var contentHeight     = $("content").getHeight();
+-    var contentWidth      = $("content").getWidth();
+-    var width = parseInt(contentWidth / 2);
+-    
+-    if ($("contentLeft"))
+-    {
+-      //Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": width + "px"});
+-      Element.setStyle("contentLeft", {"height": contentHeight + "px", "width": "55%"});
+-    }
+-    if ($("contentLeft"))
+-    {
+-      //Element.setStyle("contentRight", {"height": contentHeight + "px", "width": (width - 10) + "px"});
+-      Element.setStyle("contentRight", {"height": contentHeight + "px", "width": "44%"});
+-    }
+-    
+-    if ($("favSelector"))
+-    {   
+-      var FavSelectorHeight = $("favSelector").getHeight();
+-      var FAV_MARGIN_HEIGHT = 8;          
+-      var favViewHeight = (contentHeight - FavSelectorHeight - FAV_MARGIN_HEIGHT) * 0.95;
+-      $("favView").setStyle({"height": favViewHeight});
+-    }
+-  },
+ 
+   getPageMeasurement: function() {
+     var pageElements = ["body","#header", "#menubar", "#favSelector", "#favView", "#footer"],
+@@ -20493,11 +20430,7 @@
+     }
+   },
+   
+-  /**
+-   * Berechnet die Gr��e der Startseite f�r den Gast
+-   **/
+-  resizeGuest: function() { }   
+-  
++  resize: function() { }
```

Comment:
Why is this dummy `resize()` function still required?

Reply:
Same as above: I had bad luck removing this. My WebUI was not displaying pages correctly.

Reply:
I also stumbled upon an error that was generated by this call:

```
  /**
   * Lädt eine Seite.
   */
  enter: function(page, options)
  {
    Debug.assert(Interface.isImplemented(page, IPage), "WebUI.enter: Interface IPage not implemented");
    
    if (this.currentPage) { this.currentPage.leave(); }
    
    this.previousPage        = this.currentPage;
    this.previousPageOptions = this.currentPageOptions;
```